src/Element.cc

```diff
@@ -170,6 +201,25 @@ ParamPtr Element::CreateParam(const std::string &_key,
   return param;
 }
 
+/////////////////////////////////////////////////
+ParamPtr Element::CreateParam(const std::string &_key,
+                              const std::string &_type,
+                              const std::string &_defaultValue,
+                              bool _required,
+                              sdf::Errors &_errors,
+                              const std::string &_description)
+{
+  ParamPtr param = std::make_shared<Param>(
```

Comment:
This is duplicating code. See my comments on `AddValue`

Reply:
Since this is private, I'll go ahead and change the function to just use `sdf::Errors`.